### ISSUE TITLE
Can tf.placeholder  be used as 'model' for DeepExplainer module??

### ISSUE THREAD

**Issue by kithevans** [shap/shap#1201]
I'm trying to use DeepExplainer for my simple neural network model.

I made the model by using native tensorflow not keras.

According to the help of DeepExplainer, first argument must be input and output 

model : if framework == 'tensorflow', (input : [tf.Tensor], output : tf.Tensor)
    A pair of TensorFlow tensors (or a list and a tensor) that specifies the input and
    output of the model to be explained. Note that SHAP values are specific to a single
    output value, so the output tf.Tensor should be a single dimensional output (,1).

In my model, input X and y are claimed by tf.placeholder and feeded by during running training session like following code.

```
X = tf.compat.v1.placeholder(tf.float32, shape=[None, n_inputs], name="X")
y = tf.compat.v1.placeholder(tf.float32, shape=[None], name="y") 

with tf.compat.v1.Session() as net: 
  init = tf.compat.v1.global_variables_initializer()
  net.run(init)

  for epoch in range(n_epochs):
      for iteration in range(len(X_train_scaled) // batch_size):
          net.run(opt, feed_dict={X:X_train_scaled, y: y_train, drop_rate: 0.3, is_training: False})
```

So, I think this would work

```
background = X_train_scaled[np.random.choice(X_train_scaled.shape[0], 100, replace=False)]
  shap.DeepExplainer(([X], y),data=background,session=net)

```
but it doesn't work

```
---------------------------------------------------------------------------
InvalidArgumentError                      Traceback (most recent call last)
/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py in _do_call(self, fn, *args)
   1364     try:
-> 1365       return fn(*args)
   1366     except errors.OpError as e:

9 frames
InvalidArgumentError: You must feed a value for placeholder tensor 'y' with dtype float and shape [?]
	 [[{{node y}}]]

During handling of the above exception, another exception occurred:

InvalidArgumentError                      Traceback (most recent call last)
/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py in _do_call(self, fn, *args)
   1382                     '\nsession_config.graph_options.rewrite_options.'
   1383                     'disable_meta_optimizer = True')
-> 1384       raise type(e)(node_def, op, message)
   1385 
   1386   def _extend_graph(self):

InvalidArgumentError: You must feed a value for placeholder tensor 'y' with dtype float and shape [?]
	 [[node y (defined at <ipython-input-131-dc93b47b1c84>:23) ]]
```

Any idea what's going wrong??

**Comment by xiewenjing1170** [comment]
Have you solve the problem？I want to use DeepExplainer to interpret a Tensorflow model(.pb file). But I don't know how to load the model as tensors. If you have solved it, could you tell me? Thanks

**Comment by kithevans** [comment]
> Have you solve the problem？I want to use DeepExplainer to interpret a Tensorflow model(.pb file). But I don't know how to load the model as tensors. If you have solved it, could you tell me? Thanks

No, I haven't I just changed my code to tensorflow-keras  to use this package.
I'm sorry I can't give any help.

**Comment by alfredonuhe** [comment]
I am facing the same problem. @xiewenjing1170 in the end did you find a solution? Thanks.

**Comment by alfredonuhe** [comment]
I have found a way to run shap for my model without using keras. Instead of using the placeholder as output tensor for DeepExplainer, I used the softmax normalized predictions tensor.

Additional information: it took me 7 hours to find this fix.

**Comment by vyse8** [comment]
Hi @alfredonuhe, thanks for your suggestion as I am running into the same issue. Would you be able to provide a code example so that I can see how you are passing the tensor? Thanks.

**Comment by Nasirkhan94** [comment]
Hi @alfredonuhe, could you please explain the workaround for this issue? i am having the same issue when feeding a placeholder in TensorFlow DeepExplainer

**Comment by alfredonuhe** [comment]
Hi @vyse8 and @Nasirkhan94, for some strange reason I wasn't notified about your comments. I might be too late, but in case it helps here is a [gist example](https://gist.github.com/alfredonuhe/f925a9a3400b41a2564c55789da18d58).

**Comment by Nasirkhan94** [comment]
> Hi @vyse8 and @Nasirkhan94, for some strange reason I wasn't notified about your comments. I might be too late, but in case it helps here is a [gist example](https://gist.github.com/alfredonuhe/f925a9a3400b41a2564c55789da18d58).

Hello, 
@alfredonuhe 
I was able to fix the issue using the example you provided. It was working fine for my implementation for a while. Now, I am getting this error:
 " 1
Your TensorFlow version is newer than 2.4.0 so graph support has been removed in eager mode and some static graphs may not be supported. See PR #1483 for discussion."

The execution is stuck with the above and waiting for several hours did not resolve the issue. Any workarounds to get it resolved?
